### PR TITLE
[SD-613] Fixes redirect to paths after sign up

### PR DESCRIPTION
**Pull request by PSzyszkaSpark**
Fixes https://spark-solutions.atlassian.net/browse/SD-613

### CODE REVIEW

File: app/controllers/spree/user_registrations_controller.rb [hunk 1]
```diff
@@ -88,4 +95,14 @@ def spree_user_params
...
     stored_location_for(resource_or_scope) || account_path
   end
+
+  def redirect_to_checkout_or_account_path(resource)
+    resource_path = after_sign_up_path_for(resource)
+
+    if resource_path == spree.checkout_state_path(:address)
+      respond_with resource, location: spree.checkout_state_path(:address)
+    else
+      respond_with resource, location: account_path
```
Comment (damianlegawiec) [inline app/controllers/spree/user_registrations_controller.rb]:
```suggestion
      respond_with resource, location: spree.account_path
```

File: app/controllers/spree/user_registrations_controller.rb [hunk 2]
```diff
@@ -79,6 +79,13 @@ def after_sign_up_path_for(resource)
     after_sign_in_redirect(resource) if is_navigational_format?
   end
 
+  def after_inactive_sign_up_path_for(resource)
```
Comment (przemosk) [inline app/controllers/spree/user_registrations_controller.rb]: are You using this method somewhere ? I suppose should be here: https://github.com/spree/spree_auth_devise/pull/478/files#diff-7c1521146da1200df72dc7ea1cd3aa36R100 ?

File: app/controllers/spree/user_registrations_controller.rb [hunk 3]
```diff
@@ -88,4 +95,14 @@ def spree_user_params
   def after_sign_in_redirect(resource_or_scope)
     stored_location_for(resource_or_scope) || account_path
   end
+
+  def redirect_to_checkout_or_account_path(resource)
+    resource_path = after_sign_up_path_for(resource)
```
Comment (przemosk) [inline app/controllers/spree/user_registrations_controller.rb]: ?
```suggestion
    resource_path = after_inactive_sign_up_path_for(resource)
```

File: app/controllers/spree/user_registrations_controller.rb (hunk 2 above)
Comment (PSzyszkaSpark) [reply app/controllers/spree/user_registrations_controller.rb]: @przemosk It's used in line 36 `respond_with resource, location: after_inactive_sign_up_path_for(resource)` when user is signed in verification is required.

File: app/controllers/spree/user_registrations_controller.rb (hunk 3 above)
Comment (PSzyszkaSpark) [reply app/controllers/spree/user_registrations_controller.rb]: If I change this method than after sign up in `checkout/registration` I will be redirected to account_path and not back to `checkout/address path`

**Pull request merged**